xchange-stream-btcmarkets/src/main/java/info/bitrich/xchangestream/btcmarkets/BTCMarketsStreamingService.java

```diff
@@ -5,66 +5,82 @@
+    }
 
-  private BTCMarketsWebSocketSubscribeMessage buildSubscribeMessage() {
-    return new BTCMarketsWebSocketSubscribeMessage(
-        new ArrayList<>(subscribedOrderbooks),
-        Lists.newArrayList(CHANNEL_ORDERBOOK, CHANNEL_HEARTBEAT),
-        null,
-        null,
-        null);
-  }
+    private BTCMarketsWebSocketSubscribeMessage buildSubscribeMessage() {
+        return new BTCMarketsWebSocketSubscribeMessage(
+                new ArrayList<>(subscribedtick),
+                Lists.newArrayList(CHANNEL_ORDERBOOK, CHANNEL_HEARTBEAT, CHANNEL_TICK),
+                null,
+                null,
+                null);
+    }
 
-  @Override
-  protected String getChannelNameFromMessage(JsonNode message) {
-    final String messageType = message.get("messageType").asText();
-    if (messageType.startsWith(CHANNEL_ORDERBOOK)) {
-      return messageType + ":" + message.get("marketId").asText();
+    @Override
+    protected String getChannelNameFromMessage(JsonNode message) {
+        final String messageType = message.get("messageType").asText();
+        if (messageType.startsWith(CHANNEL_TICK)) {
+            return messageType + ":" + message.get("marketId").asText();
+        } else if (messageType.startsWith(CHANNEL_ORDERBOOK)) {
+            return messageType + ":" + message.get("marketId").asText();
+        } else
+            return messageType;
     }
-    return messageType;
-  }
 
-  @Override
-  public String getSubscribeMessage(String channelName, Object... args) throws IOException {
-    if (CHANNEL_ORDERBOOK.equals(channelName)) {
-      subscribedOrderbooks.add(args[0].toString());
-      LOG.debug("Now subscribed to orderbooks {}", subscribedOrderbooks);
-      return objectMapper.writeValueAsString(buildSubscribeMessage());
-    } else {
-      throw new IllegalArgumentException(
-          "Can't create subscribe messsage for channel " + channelName);
+    @Override
+    public String getSubscribeMessage(String channelName, Object... args) throws IOException {
+        if (CHANNEL_TICK.equals(channelName)) {
+            subscribedtick.add(args[0].toString());
+            LOG.debug("Now subscribed to Ticker {}", subscribedtick);
+            return objectMapper.writeValueAsString(buildSubscribeMessage());
+        } else if (CHANNEL_ORDERBOOK.equals(channelName)) {
+            subscribedOrderbooks.add(args[0].toString());
+            LOG.debug("Now subscribed to Orderbook {}", subscribedOrderbooks);
+            return objectMapper.writeValueAsString(buildSubscribeMessage());
+        } else {
+            throw new IllegalArgumentException(
+                    "Can't create subscribe messsage for channel " + channelName);
+        }
     }
-  }
 
-  public String getSubscriptionUniqueId(String channelName, Object... args) {
-    if (CHANNEL_ORDERBOOK.equals(channelName)) {
-      return channelName + ":" + args[0].toString();
+    public String getSubscriptionUniqueId(String channelName, Object... args) {
```

Comment:
pedantic but this would lend itself nicely to a switch statement

Reply:
Done